src/classes/ALLO_MakeDefaultAllocations_BATCH.cls

```diff
@@ -134,11 +134,11 @@ public class ALLO_MakeDefaultAllocations_BATCH implements Database.Batchable<sOb
         //No error handling necessary, because the only place this is called from is the widget that displays the batch
         //process status in the UI.
         if (!listAlloForInsert.isEmpty()) {
-            ALLO_Allocations_TDTM.resetFlags(true);
+            ALLO_ProcessControl.setRecursionFlag(true);
```

Comment:
I'd prefer seeing the explicit cleanup, to avoid people making assumptions about execution context, and accidentally following this pattern in non-batch scenarios.